Trie/Trie/Trie.cs

```diff
@@ -0,0 +1,214 @@
+                previousSymbol = last.Value;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        if (isPreviousLeaf)
+        {
+            this.Root.Children.Remove(previousSymbol);
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Determines whether the Trie contains the specified string.
+    /// </summary>
+    /// <param name="element">The string to locate in the Trie.</param>
+    /// <returns>
+    /// <c>true</c> if the Trie contains the string;
+    /// <c>false</c> otherwise.
+    /// </returns>
+    /// <exception cref="ArgumentException">
+    /// Thrown when the input string is null or empty.
+    /// </exception>
+    public bool Contains(string element)
+    {
+        ArgumentNullException.ThrowIfNull(element, nameof(element));
+        if (element.Length == 0)
+        {
+            throw new ArgumentException("String cannot be empty", nameof(element));
+        }
+
+        var (endPrefixNode, nextPosition, pathStack) = this.FindLongestPrefix(element, false);
+
+        return nextPosition == element.Length && endPrefixNode.IsTerminal == true;
+    }
+
+    /// <summary>
+    /// Counts how many complete strings in the Trie start with the specified prefix.
+    /// </summary>
+    /// <param name="prefix">The prefix to search for.</param>
+    /// <returns>
+    /// The number of complete strings that start with the prefix.
+    /// Returns 0 if the prefix doesn't exist in the Trie.
+    /// </returns>
+    /// <exception cref="ArgumentException">
+    /// Thrown when the input string is null or empty.
+    /// </exception>
+    public int HowManyStartsWithPrefix(string prefix)
+    {
+        ArgumentNullException.ThrowIfNull(prefix, nameof(prefix));
+        if (prefix.Length == 0)
+        {
+            throw new ArgumentException("Prefix cannot be empty", nameof(prefix));
+        }
+
+        var (endPrefixNode, nextPosition, pathStack) = this.FindLongestPrefix(prefix, false);
+
+        return nextPosition == prefix.Length ? endPrefixNode.Children.Count : 0;
+    }
+
+    private static Node CreateSuffix(string str)
+    {
+        ArgumentNullException.ThrowIfNull(str, nameof(str));
+        if (str.Length == 0)
+        {
+            throw new ArgumentException("String cannot be empty", nameof(str));
+        }
+
+        Node source = new Node(str[0]);
+        Node previous = source;
+
+        for (int i = 1; i < str.Length; i++)
+        {
+            Node newNode = new Node(str[i]);
+            previous.Children[str[i]] = newNode;
+            previous = newNode;
+        }
+
+        previous.IsTerminal = true;
+        return source;
+    }
+
+    private (Node EndPrefixNode, int NextPosition, Stack<Node>? PathStack) FindLongestPrefix(string element, bool needPathStack)
+    {
+        int currentPosition = 0;
+        Node currentNode = this.Root;
+        Stack<Node> stack = new Stack<Node>();
```

Comment:
```suggestion
        Stack<Node> stack = new();
```
Даже на Java уже так не пишут, хотя Java последняя сдалась